### PR TITLE
feat: add style prop to toggle visibility in component registration [SPA-2371]

### PR DESCRIPTION
**Pull request by ChidinmaOrajiaku**
## Purpose
Introduce a cfVisibility style prop variable to the default style definitions

## Approach

- Add ` cfVisibility` to cf component default style definitions
- For custom components, add ` cfVisibility` default variable on component registration

### CODE REVIEW

File: packages/core/src/definitions/styles.ts [hunk 1]
```diff
@@ -56,6 +56,13 @@ export const builtInStyles: VariableDefinitions = {
     defaultValue: 'center',
     displayName: 'Horizontal alignment',
   },
+  cfVisibility: {
+    displayName: 'Visibility toggle',
+    type: 'Boolean',
+    group: 'style',
+    defaultValue: false,
+    description: 'To enable visibility toggle in patterns',
```
Comment (Spring3) [inline packages/core/src/definitions/styles.ts:64]: Is it only going to apply to the patterns or to all registered components? 🤔

Comment (ChidinmaOrajiaku) [reply packages/core/src/definitions/styles.ts]: Thanks for pointing that out

File: packages/core/src/definitions/styles.ts [hunk 2]
```diff
@@ -56,6 +56,13 @@ export const builtInStyles: VariableDefinitions = {
     defaultValue: 'center',
     displayName: 'Horizontal alignment',
   },
+  cfVisibility: {
+    displayName: 'Visibility toggle',
+    type: 'Boolean',
+    group: 'style',
+    defaultValue: false,
```
Comment (Chaoste) [inline packages/core/src/definitions/styles.ts:63]:
```suggestion
    defaultValue: true,
```
I thought nodes should be visible, per default

File: packages/core/src/definitions/styles.ts [hunk 3]
```diff
@@ -56,6 +56,13 @@ export const builtInStyles: VariableDefinitions = {
...
+    description: 'To enable visibility toggle in components',
```
Comment (Chaoste) [inline packages/core/src/definitions/styles.ts:64]: [optional] (not sure if this is of any value but I propose aligning it with the other descriptions in here)
```suggestion
    description: 'The visibility of the component',
```

**Review by Chaoste — CHANGES_REQUESTED** [review]
It's all minor but I had some thoughts. LMK what you think and if I should give it a second look :)

**Review by Spring3 — APPROVED** [review]
Agree with changes requested by @Chaoste . I don't have anything more to add :)

**Review by Chaoste — APPROVED** [review]

**Pull request merged**